src/tlo/methods/wasting.py

```diff
@@ -1,41 +1,1947 @@
+
+                assert wasted_days >= get_min_length(recov_how, person_id, whz), \
+                    (f" The {person_id=} is wasted {wasted_days=} < minimal expected length= "
+                     f"{get_min_length(recov_how, person_id, whz)} days when {recov_opt=}.")
+                self.module.wasting_length_tracker[age_group][recov_opt].append(wasted_days)
+
+                df.at[person_id, 'un_WHZ_category'] = 'WHZ>=-2'
+                df.at[person_id, 'un_am_MUAC_category'] = '[115-125)mm'
+
+            if mam_classification == 'mam_by_muac_and_whz':
+                df.at[person_id, 'un_WHZ_category'] = '-3<=WHZ<-2'
+                df.at[person_id, 'un_am_MUAC_category'] = '[115-125)mm'
+
+            if mam_classification == 'mam_by_whz_only':
+                df.at[person_id, 'un_WHZ_category'] = '-3<=WHZ<-2'
+                df.at[person_id, 'un_am_MUAC_category'] = '>=125mm'
+
+        # Update all other properties equally
+        df.at[person_id, 'un_clinical_acute_malnutrition'] = 'MAM'
+        df.at[person_id, 'un_am_nutritional_oedema'] = False
+        df.at[person_id, 'un_sam_with_complications'] = False
+        df.at[person_id, 'un_sam_death_date'] = pd.NaT # death is cancelled if was scheduled
+        df.at[person_id, 'un_am_tx_start_date'] = pd.NaT
+        # Start without treatment, treatment will be applied with HSI if care sought
+        df.at[person_id, 'un_am_treatment_type'] = 'none'
+
+        # this will clear all wasting symptoms (applicable for SAM, not MAM)
+        self.sim.modules["SymptomManager"].clear_symptoms(
+            person_id=person_id, disease_module=self.module
+        )
+
+        outcome_date = self.module.date_of_outcome_for_untreated_wasting(whz_category='-3<=WHZ<-2')
+        progression_severe_wasting_bool = self.module.wasting_models.severe_wasting_progression_lm.predict(
+            df.loc[[person_id]], rng=rng
+        )
+
+        # natural history (if not treated)
+        if df.at[person_id, 'un_WHZ_category'] == '-3<=WHZ<-2' and progression_severe_wasting_bool:
+            self.sim.schedule_event(Wasting_ProgressionToSevere_Event(self.module, person_id), outcome_date)
+        else:
+            self.sim.schedule_event(Wasting_FullRecovery_Event(self.module, person_id), outcome_date)
+
+class Wasting_InitiateGrowthMonitoring(Event, PopulationScopeEventMixin):
+    # TODO: maybe will be updated to integrate monitoring of < 1y old in epi module, and on birth schedule to be
+    #  monitored within wasting module only when > 1y old
+    """
+    Event that schedules HSI_Wasting_GrowthMonitoring for all under-5 children for a random day within the age-dependent
+    frequency.
+    """
+
+    def __init__(self, module):
+        """Runs only once, when simulation is initiated.
+        :param module: the module that created this event
+        """
+        super().__init__(module)
+        assert isinstance(module, Wasting)
+
+    def apply(self, population):
```

Comment:
why is this an event, rather  than a function that is called at initialise_simulation?